packages/deployment-service/src/dns/devops.pr.provider.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@nestjs/common'
+import { PipelineEntity } from '../entities/pipeline.entity'
+import { GithubPullRequestProvider } from '../github/github-pull-request-provider'
+import { load, dump } from 'js-yaml'
+
+export type DnsRecords = {
+  name: string
+  value: string
+  type: 'CNAME'
+}
+
+type DevopsDnsRecord = {
+  name: string
+  hosted_zone_identifier: 'reapit.cloud'
+  type: 'CNAME'
+  ttl: 300
+  records: string[]
+}
+
+type DevopsDnsRecordConfig = {
+  config: {
+    'route53:records': DevopsDnsRecord[]
+  }
+}
+
+@Injectable()
+export class DevopsPrProvider {
+  private static repositoryOwner: string = 'reapit-global'
+  private static repositoryName: string = 'uk-devops-infrastructure-core-cdk'
+  private static repositoryFile: string = 'pulumi_projects/route53/Pulumi.route53.main.yaml'
+  constructor(private readonly githubPrProvider: GithubPullRequestProvider) {}
+
+  async createPR(cnames: DnsRecords[], pipeline: PipelineEntity) {
+    const file = await this.githubPrProvider.obtainFile({
+      owner: DevopsPrProvider.repositoryOwner,
+      filePath: DevopsPrProvider.repositoryFile,
+      repository: DevopsPrProvider.repositoryName,
+    })
+
+    if (!('content' in file.data)) throw new Error('github file not correct')
+
+    const content = Buffer.from(file.data.content, 'base64').toString('utf-8')
+
+    const yml = load(content) as DevopsDnsRecordConfig
+
+    console.log('yml', yml)
+
+    cnames.forEach((cname) => {
+      yml.config['route53:records'].push({
+        name: cname.name,
+        hosted_zone_identifier: 'reapit.cloud', // has to be reapit.cloud
+        ttl: 300,
+        records: [cname.value],
+        type: cname.type,
+      })
+    })
+
+    const updatedFile = dump(yml)
+
+    await this.githubPrProvider.createPullRequest({
```

Comment:
We probably need some idea of when this fails. Can we catch a failure and generate some kind of alert off the back of it? Looks like we might risk silent failures here

Reply:
We could? Can you spec that out for me please.